### PR TITLE
Prevent vars premature templating

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

core

### CODE REVIEW

**Review by mkrizek — APPROVED** [review]

**Comment by samdoran** [comment]
I believe it was discussed with @sivel that this can be closed.

**Comment by bcoca** [comment]
rediscussed, now using correct test

File: test/integration/targets/var_templating/runme.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env bash
+
+set -eux
+
+# this should succeed since we override the undefined variable
+ansible-playbook undefined.yml -i inventory -v "$@" -e '{"mytest": False}'
```
Comment (mkrizek) [inline test/integration/targets/var_templating/runme.sh:6]: It should succeed even if we don't override the undefined variable.

Comment (bcoca) [reply test/integration/targets/var_templating/runme.sh:6]: yes, that test is below

**Comment by mkrizek** [comment]
shipit

**Pull request merged**